web3/types.py

```diff
@@ -48,38 +46,37 @@
     "name": str,
     "type": str,
     "indexed": bool,
-})
+}, total=False)
 
-ENS = NewType("ENS", str)
 
 ABIEvent = TypedDict("ABIEvent", {
-    "type": "event",
+    "type": Literal["event"],
     "name": str,
-    "inputs": Sequence[ABIEventParams],
+    "inputs": Sequence["ABIEventParams"],
     "anonymous": bool,
-})
+}, total=False)
 
 
-ABIFunctionComponents = TypedDict("ABIFunctionParams", {
+ABIFunctionComponents = TypedDict("ABIFunctionComponents", {
     "name": str,
     "type": str,
-    "components": Sequence["ABIFunctionComponents"],
+    "components": Sequence[Any],
```

Comment:
This is better typed as it was originally, `Sequence["ABIFunctionComponents"]` - but it seems like mypy doesn't allow for recursive definitions. Is there a better solution that I'm missing besides using `Any`?

Reply:
I think typing as `Any` and linking the the relevant mypy issue about recursive types.